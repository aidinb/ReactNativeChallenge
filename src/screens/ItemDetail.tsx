import React from 'react';
import { View, ScrollView, StyleSheet, ViewStyle } from 'react-native';
import { observer } from 'mobx-react';
import { COLORS } from '../styles';
import { useStores } from '../stores';
import DetailItem from '../components/DetailItem';
import Loading from '../components/Loading';
import ErrorMessage from '../components/ErrorMessage';

const ItemDetail: React.FC = () => {
    const { mainStore } = useStores();
    const { user, loading, error } = mainStore;

    return (
        <ScrollView contentContainerStyle={styles.scrollContainer}>
            {!loading && !error && user && (
                <View style={styles.container}>
                    {/* User Details */}
                    <DetailItem title="Name" value={user.name} />
                    <DetailItem title="Email" value={user.email} />
                    <DetailItem title="Website" value={user.website} />
                    <DetailItem title="Phone" value={user.phone} />

                    {/* Address Details */}
                    <DetailItem title="Street" value={user.address?.street} />
                    <DetailItem title="Suite" value={user.address?.suite} />
                    <DetailItem title="City" value={user.address?.city} />
                    <DetailItem title="Zipcode" value={user.address?.zipcode} />

                    {/* Company Details */}
                    <DetailItem title="Company Name" value={user.company?.name} />
                    <DetailItem title="CatchPhrase" value={user.company?.catchPhrase} />
                    <DetailItem title="Business" value={user.company?.bs} />
                </View>
            )}

            {loading && <Loading />}
            {error && <ErrorMessage message={error} />}
        </ScrollView>
    );
};

const styles = StyleSheet.create({
    scrollContainer: {
        paddingBottom: 20,
    } as ViewStyle,
    container: {
        backgroundColor: COLORS.white,
        flex: 1,
    } as ViewStyle,
});

export default observer(ItemDetail);
