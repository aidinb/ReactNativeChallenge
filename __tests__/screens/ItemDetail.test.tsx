import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import ItemDetail from '../../src/screens/ItemDetail.tsx'; // Import ItemDetail from the correct path


describe('ItemDetail Screen', () => {
    it('renders correctly and displays user details', async () => {
        const { toJSON, getByText } = render(<ItemDetail />);

        // Check if the component renders correctly
        expect(toJSON()).toMatchSnapshot();

        // Check if user details are displayed
        await waitFor(() => {
            expect(getByText('Name')).toBeTruthy();
            expect(getByText('John Doe')).toBeTruthy();
            expect(getByText('Email')).toBeTruthy();
            expect(getByText('john@example.com')).toBeTruthy();
            expect(getByText('Website')).toBeTruthy();
            expect(getByText('www.example.com')).toBeTruthy();
            expect(getByText('Phone')).toBeTruthy();
            expect(getByText('123-456-7890')).toBeTruthy();
            expect(getByText('Street')).toBeTruthy();
            expect(getByText('123 Main St')).toBeTruthy();
            expect(getByText('Suite')).toBeTruthy();
            expect(getByText('Apt 4B')).toBeTruthy();
            expect(getByText('City')).toBeTruthy();
            expect(getByText('Sample City')).toBeTruthy();
            expect(getByText('Zipcode')).toBeTruthy();
            expect(getByText('12345')).toBeTruthy();
            expect(getByText('Company Name')).toBeTruthy();
            expect(getByText('Sample Company')).toBeTruthy();
            expect(getByText('CatchPhrase')).toBeTruthy();
            expect(getByText('We do things!')).toBeTruthy();
            expect(getByText('Business')).toBeTruthy();
            expect(getByText('Business Solutions')).toBeTruthy();
        });
    });
});
