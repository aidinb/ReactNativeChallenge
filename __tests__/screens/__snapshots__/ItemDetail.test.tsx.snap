// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail Screen renders correctly and displays user details 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "paddingBottom": 20,
    }
  }
>
  <View>
    <View
      style={
        {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        style={
          {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          Name
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          John Doe
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          Email
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          john@example.com
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          Website
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          www.example.com
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          Phone
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          123-456-7890
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          Street
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          123 Main St
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          Suite
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Apt 4B
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          City
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Sample City
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          Zipcode
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          12345
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          Company Name
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Sample Company
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          CatchPhrase
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          We do things!
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          Business
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Business Solutions
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
